tests/ui/match/ref_pat_eat_one_layer_2024/ref_pat_eat_one_layer_2024_fail.rs

```diff
@@ -50,6 +52,10 @@ pub fn main() {
     let &mut &mut &mut &mut _ = &mut &&&&mut &&&mut &mut 0;
     //~^ ERROR: mismatched types
 
+    if let Some(&mut _) = &mut Some(&0) {
+        //[structural]~^ ERROR
+    }
```

Comment:
This one will start working when we change to the revised (extended) *Rule 4*.

Reply:
Yes, hence why it is marked as failing in the `structural` variant only (not `both` which is the revised rule).

Reply:
Makes sense.